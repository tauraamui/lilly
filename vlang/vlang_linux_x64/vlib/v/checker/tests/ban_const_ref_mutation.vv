struct Foobar {
mut:
	foo int
}

const foobar_const = Foobar{123}

fn main() {
	println('foobar_const.foo: ${foobar_const.foo}') // foobar_const.foo: 123
	mut foobars := []&Foobar{}
	foobars << &foobar_const
	foobars[0].foo = 456
	println('foobar_const.foo: ${foobar_const.foo}') // foobar_const.foo: 456
}
