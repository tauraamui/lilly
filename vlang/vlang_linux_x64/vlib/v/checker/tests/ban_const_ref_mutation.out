vlib/v/checker/tests/ban_const_ref_mutation.vv:11:14: error: cannot have mutable reference to const `foobar_const`
    9 |     println('foobar_const.foo: ${foobar_const.foo}') // foobar_const.foo: 123
   10 |     mut foobars := []&Foobar{}
   11 |     foobars << &foobar_const
      |                 ~~~~~~~~~~~~
   12 |     foobars[0].foo = 456
   13 |     println('foobar_const.foo: ${foobar_const.foo}') // foobar_const.foo: 456
